cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/TemplatesUtilsClassesUtil.java

```diff
@@ -0,0 +1,293 @@
+
+            URI uri = URI.create(contextConfigurationLocation.toString());
+            FileSystem fs;
+            try {
+                fs = FileSystems.getFileSystem(uri);
+            } catch (FileSystemNotFoundException e) {
+                fs = FileSystems.newFileSystem(uri, env);
+            }
+            Paths.get(uri);
+
+            try {
+                // Get the URI of the jar from the URL of the contained context.xml
+                URI jarUri = URI.create(contextConfigurationLocation.toString().split("!")[0]);
+                FileSystem jarfs = FileSystems.getFileSystem(jarUri);
+
+                // walk the jar file
+                LOG.debug("Searching for classes in " + jarUri.toString());
+                Files.walkFileTree(jarfs.getPath("/"), new SimpleFileVisitor<Path>() {
+
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        if (file.toString().endsWith(".class")) {
+                            LOG.debug("    * Found class file " + file.toString());
+                            // remove the leading '/' and the trailing '.class'
+                            String fileName = file.toString().substring(1, file.toString().length() - 6);
+                            // replace the path separator '/' with package separator '.' and add it to the
+                            // list of found files
+                            foundClasses.add(fileName.replace("/", "."));
+                        }
+                        return FileVisitResult.CONTINUE;
+                    }
+
+                    @Override
+                    public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                        // Log errors but do not throw an exception
+                        LOG.warn("visitFileFailed @ " + exc);
+                        return FileVisitResult.CONTINUE;
+                    }
+                });
+            } catch (IOException e) {
+                LOG.error("Could not read templates jar file " + e);
+            }
+            for (String className : foundClasses) {
+                try {
+                    result.add(inputClassLoader.loadClass(className));
+                } catch (ClassNotFoundException e) {
+                    LOG.warn("Could not load " + className + " from classpath", e);
+                }
+            }
+        } else {
+            LOG.info("Processing configuration folder " + templateRoot.toString());
+            LOG.debug("Searching for classes ...");
+            final List<Path> foundPaths = new LinkedList<>();
+
+            try {
+                Files.walkFileTree(templateRoot, new SimpleFileVisitor<Path>() {
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        if (file.toString().endsWith(".class")) {
```

Comment:
I think path api also provides an getExtension api so you don't need tostring resulting in bad performance

Reply:
adjusted.